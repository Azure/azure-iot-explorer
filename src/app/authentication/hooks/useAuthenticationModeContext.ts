/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';

export enum AuthenticationMode {
    connectionString,
    roleBased
}

export const useAuthenticationMode = (): AuthenticationModeContextType => {
    const [ authenticationMode, setAuthenticationMode ] = React.useState<AuthenticationMode>(AuthenticationMode.connectionString);
    return {
        authenticationMode,
        setAuthenticationMode
    };
};

export interface AuthenticationModeContextType {
    authenticationMode: AuthenticationMode;
    setAuthenticationMode(authenticationMode: AuthenticationMode): void;
}

export const AuthenticationModeContext = React.createContext<AuthenticationModeContextType>({
    authenticationMode: AuthenticationMode.connectionString,
    setAuthenticationMode: () => {} // tslint:disable-line: no-empty
});
export const useAuthenticationModeContext = () => React.useContext<AuthenticationModeContextType>(AuthenticationModeContext);
