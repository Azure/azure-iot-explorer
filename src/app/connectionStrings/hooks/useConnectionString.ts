import { useState } from 'react';
import { ConnectionStringType } from '../context/connectionStringContext';

export const useConnectionString = (): ConnectionStringType => {
    const [connectionString, setConnectionString] = useState<string>('');
    return [{connectionString}, {setConnectionString}];
};
