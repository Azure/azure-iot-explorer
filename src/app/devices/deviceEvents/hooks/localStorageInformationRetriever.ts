import * as React from 'react';
import { CONNECTION_STRING_NAME_LIST } from '../../../constants/browserStorage';
import { getConnectionInfoFromConnectionString } from '../../../api/shared/utils';
import { getActiveConnectionString } from '../../../shared/utils/hubConnectionStringHelper';

interface LocalStorageInformation {
    hubConnectionString: string;
    hostName: string;
}

export const getHubInformationFromLocalStorage = (): LocalStorageInformation => {
    const [ hubConnectionString, setHubConnectionString ] = React.useState<string>('');
    const [ hostName, setHostName ] = React.useState<string>('');
    const connectionStrings = localStorage.getItem(CONNECTION_STRING_NAME_LIST);
    const connectionString = getActiveConnectionString(connectionStrings);

    React.useEffect(() => {
        setHubConnectionString(connectionString);
        setHostName(getConnectionInfoFromConnectionString(connectionString).hostName);
    },              [connectionString]);

    return {hubConnectionString, hostName};
};
