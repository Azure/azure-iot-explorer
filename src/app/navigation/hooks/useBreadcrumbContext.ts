/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { BreadcrumbEntry } from '../model';

export interface BreadcrumbContextType {
    stack: BreadcrumbEntry[];
    registerEntry(entry: BreadcrumbEntry): void;
    unregisterEntry(entry: BreadcrumbEntry): void;
}

export const BreadcrumbContext = React.createContext<BreadcrumbContextType>({
    registerEntry: () => {}, // tslint:disable-line: no-empty
    stack: [],
    unregisterEntry: () => {} // tslint:disable-line: no-empty
});

export const useBreadcrumbContext = () => React.useContext<BreadcrumbContextType>(BreadcrumbContext);
