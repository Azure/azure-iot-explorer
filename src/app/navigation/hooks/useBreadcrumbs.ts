/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { BreadcrumbEntry } from '../model';

export interface UseBreadcrumbsResult {
    stack: BreadcrumbEntry[];
    registerEntry(entry: BreadcrumbEntry): void;
    unregisterEntry(entry: BreadcrumbEntry): void;
}

export const useBreadcrumbs = (): UseBreadcrumbsResult => {
    const [stack, setStack] = React.useState<BreadcrumbEntry[]>([]);

    const registerEntry = React.useCallback((entry: BreadcrumbEntry) => {
        setStack(currentState => {
            const newStack = [...currentState.filter(s => entry.path !== s.path), entry];
            const sortedStack = newStack.sort((a, b) => a.path.length - b.path.length);

            return sortedStack;
        });
    }, []); // tslint:disable-line: align

    const unregisterEntry = React.useCallback((entry: BreadcrumbEntry) => {
        setStack(currentState => {
            const newStack = currentState.filter(s => entry.path !== s.path);
            return newStack;
        });
    }, []); // tslint:disable-line: align

    return {stack, registerEntry, unregisterEntry};
};
