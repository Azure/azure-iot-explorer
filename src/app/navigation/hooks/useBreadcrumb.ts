/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { BreadcrumbEntry } from '../model';
import { BreadcrumbContextType } from './useBreadcrumbContext';

export const useBreadcrumbs = (): BreadcrumbContextType => {
    const [stack, setStack] = React.useState<BreadcrumbEntry[]>([]);

    const registerEntry = React.useCallback((entry: BreadcrumbEntry) => {
        setStack(currentState => {
            const newStack = [...currentState.filter(s => entry.path !== s.path), entry];
            const sortedStack = newStack.sort((a, b) => a.path.length - b.path.length);

            return sortedStack;
        });
    }, []); // tslint:disable-line: align

    const unregisterEntry = React.useCallback((entry: BreadcrumbEntry) => {
        setStack(currentState => {
            const newStack = currentState.filter(s => entry.path !== s.path);
            return newStack;
        });
    }, []); // tslint:disable-line: align

    return {stack, registerEntry, unregisterEntry};
};
