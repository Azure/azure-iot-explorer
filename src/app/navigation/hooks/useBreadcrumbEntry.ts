/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useRouteMatch } from 'react-router-dom';
import { useBreadcrumbContext } from './useBreadcrumbContext';

export interface UseBreadcrumbEntryProps {
    disableLink?: boolean;
    name: string;
    suffix?: string;
}

export const useBreadcrumbEntry = ({ name, disableLink, suffix }: UseBreadcrumbEntryProps) => {
    const { registerEntry, unregisterEntry } = useBreadcrumbContext();
    const { path, url } = useRouteMatch();

    React.useEffect(() => {
        registerEntry({ name, disableLink, path, suffix, url });
        return () => {
            unregisterEntry({ name, disableLink, path, suffix, url });
        };
    }, [ suffix, name, disableLink ]); // tslint:disable-line: align
};
