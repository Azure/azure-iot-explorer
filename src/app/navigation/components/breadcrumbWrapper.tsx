/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useBreadcrumbEntry, UseBreadcrumbEntryProps } from '../hooks/useBreadcrumbEntry';

export type BreadcrumbWrapperProps = UseBreadcrumbEntryProps;

export const BreadcrumbWrapper: React.FC<BreadcrumbWrapperProps> = props => {
    useBreadcrumbEntry(props);
    return <>{props.children}</>;
};
