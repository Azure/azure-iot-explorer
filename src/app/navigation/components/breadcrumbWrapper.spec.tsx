/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { shallow } from 'enzyme';
import { BreadcrumbWrapper } from './breadcrumbWrapper';
import * as BreadcrumbEntry from '../hooks/useBreadcrumbEntry';

describe('BreadcrumbWrapper', () => {
    it('matches snapshot', () => {
        jest.spyOn(BreadcrumbEntry, 'useBreadcrumbEntry').mockImplementation(() => {});
        expect(shallow(
            <BreadcrumbWrapper
                name={'name'}
            />
        )).toMatchSnapshot();
    });
});