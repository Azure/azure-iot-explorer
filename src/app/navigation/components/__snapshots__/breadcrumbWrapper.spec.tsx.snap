// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbWrapper matches snapshot 1`] = `<Fragment />`;
